webpack.config.js

```diff
@@ -271,7 +271,7 @@ if ( config.isEnabled( 'webpack/persistent-caching' ) ) {
 if ( shouldMinify ) {
 	webpackConfig.plugins.push(
 		new UglifyJsPlugin( {
-			cache: true,
+			cache: 'docker' !== process.env.CONTAINER,
```

Comment:
Can you explain this line?
Docker cannot use the uglifyjs cache?

Reply:
@samouri would we _want_ docker to use the `uglifyjs` cache? wouldn't that introduce non-determinism we're trying to prevent?

Reply:
I tried to provide a good description in the commit message for d0ef40d1e20c0f9abeeeda91cf5479c082473e6d:

> Uglify produces lots of cache files. By default, these are located in
`node_modules`. Persisting uglify cache in Docker image serves no
purpose. If files have changed, the layer will not be reused and cache
is irrelevant. If files remain unchanged, the entire layer will be
cached, and the uglify cache is again irrelevant.

In other words…

This caused problems with the approach in this PR because it produces a bunch of new files in `node_modules`, which then would need to be `chown`ed again..

What's the point of these cache files? They're persisted on disk so that repeated work can be skipped. Cache files are to improve _subsequent runs_, but Docker doesn't work like that.

```
pre-build layer                                    post-build layer
     |                                                  |
     |------build Calypso (produce cache files)-------->|
     |                                                  |
```

If we were to continue rebuilding at the `post-build layer`, those cache files would be great. However, because of the way Docker works, if there are new source files, we'll always start at `pre-build layer` without any cache.

Succinctly, if source files aren't changed, the cache serves no purpose, if they are changed, we won't have access to the cache in Docker anways.

If my reasoning isn't off, cached files only serve to complicate the approach proposed in this 
PR and add some unnecessary size to the resulting image.

Reply:
I missed the commit message. Thank you so much for this kickass explanation!
Makes 1000% sense 👍